tests/vmi_multus_test.go

```diff
@@ -375,32 +375,50 @@ var _ = Describe("[Serial]Multus", func() {
-				vmiTwo := createVMIOnNode(interfaces, networks)
-				tests.WaitUntilVMIReady(vmiTwo, console.LoginToAlpine)
+				vmiTwo := libvmi.NewFedora(
+					libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+					libvmi.WithNetwork(v1.DefaultPodNetwork()),
+					libvmi.WithInterface(linuxBridgeInterface),
+					libvmi.WithNetwork(&linuxBridgeNetwork),
+					libvmi.WithCloudInitNoCloudUserData(tests.GetGuestAgentUserData(), false),
+					libvmi.WithCloudInitNoCloudNetworkData(cloudInitNetworkDataWithStaticIPsByDevice("eth1", "10.1.1.2/24"), false))
+				vmiTwo = tests.StartVmOnNode(vmiTwo, nodes.Items[0].Name)
 
 				By("Creating another VM with custom MAC address on its Linux bridge CNI interface.")
-				interfaces[linuxBridgeIfIdx].MacAddress = customMacAddress
-				vmiOne := createVMIOnNode(interfaces, networks)
-				tests.WaitUntilVMIReady(vmiOne, console.LoginToAlpine)
-
-				By("Configuring static IP address to the Linux bridge interface.")
-				Expect(configInterface(vmiOne, "eth0", "10.1.1.1/24")).To(Succeed())
-				Expect(configInterface(vmiTwo, "eth0", "10.1.1.2/24")).To(Succeed())
+				linuxBridgeInterfaceWithCustomMac := linuxBridgeInterface
+				linuxBridgeInterfaceWithCustomMac.MacAddress = customMacAddress
+				vmiOne := libvmi.NewFedora(
+					libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+					libvmi.WithNetwork(v1.DefaultPodNetwork()),
+					libvmi.WithInterface(linuxBridgeInterfaceWithCustomMac),
+					libvmi.WithNetwork(&linuxBridgeNetwork),
+					libvmi.WithCloudInitNoCloudUserData(tests.GetGuestAgentUserData(), false),
+					libvmi.WithCloudInitNoCloudNetworkData(cloudInitNetworkDataWithStaticIPsByMac(customMacAddress, "10.1.1.1/24"), false))
+				vmiOne = tests.StartVmOnNode(vmiOne, nodes.Items[0].Name)
+
+				tests.WaitUntilVMIReady(vmiOne, console.LoginToFedora)
+				tests.WaitAgentConnected(virtClient, vmiOne)
+
+				tests.WaitUntilVMIReady(vmiTwo, console.LoginToFedora)
+				tests.WaitAgentConnected(virtClient, vmiTwo)
 
 				By("Verifying the desired custom MAC is the one that were actually configured on the interface.")
-				ipLinkShow := fmt.Sprintf("ip link show eth0 | grep -i \"%s\" | wc -l\n", customMacAddress)
-				err = console.SafeExpectBatch(vmiOne, []expect.Batcher{
-					&expect.BSnd{S: ipLinkShow},
-					&expect.BExp{R: "1"},
-				}, 15)
-				Expect(err).ToNot(HaveOccurred())
+				Eventually(func() string {
+
+					vmiOne, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(vmiOne.Name, &metav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred(), "should find vmiOne to check nic0 MAC at Status")
+
+					for _, iface := range vmiOne.Status.Interfaces {
+						if iface.Name == linuxBridgeInterfaceWithCustomMac.Name {
+							return iface.MAC
+						}
+					}
+					return ""
+				}).Should(Equal(customMacAddress), "should set linux bridge interface with the custom MAC address")
```

Comment:
> BUT, all the above may be irrelevant as it seems we read the mac address for non-SRIOV and non-custom (created by user in the VM itself) from the domain spec and not from the GA. Only in case of SRIOV or an interface that was created by the user inside the VM, we report the mac from the GA.
Therefore, I think we do not need GA at all in this test and there is no need to wait with Eventually. Have you seen this needed while testing?

Then we have an issue here, the original version of this test looks inside the VMI to check the mac address, but the version at this PR check the Status I though status was being set by guest agent, but looks like it's not ? then we cannot use status here.

Reply:
IMO we can, as we are not tracking of libvirt is working well or not here.
On the other hand, we should make sure to overwrite the mac address if it is different from the one reported (or add a warning about it).
Sound like an issue we have not covered in both production and tests (but I may be wrong here).